#!/usr/bin/env python
PACKAGE = "ur5_control"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("TAU_0", double_t, 0, "DMP time constant", 1.0, 0.1, 10.0)
gen.add("ALPHA_X", double_t, 0, "Spring constant", 20.0, 0.0, 100.0)
gen.add("ALPHA_S", double_t, 0, "Alpha_s", 4.0, 0.0, 20.0)
gen.add("ALPHA_G", double_t, 0, "Goal lowpass", 20.0, 0.0, 100.0) #with a value of 20, in 0.3sec the new goal become the DMP's goal
gen.add("ALPHA_D", double_t, 0, "Hand distance lowpass", 20.0, 0.0, 100.0)
gen.add("ALPHA_E", double_t, 0, "Tracking error lowpass", 20.0, 0.0, 100.0)
gen.add("K_OrntDyn", double_t, 0, "OrntDyn gain", 0.4, 0.0, 20.0)
gen.add("fc_OrntDyn", double_t, 0, "OrntDyn low-pass cutting frequency", 8.0, 0.01, 20.0)

exit(gen.generate(PACKAGE, "ur5_control", "DMPParams"))
